packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -95,3 +95,123 @@ class HttpRouteWholePathMatch extends HttpRoutePathMatch {
     };
   }
 }
+
+/**
+ * The type returned from the `bind()` method in {@link HttpGatewayRoutePathMatch}.
+ */
+export interface HttpGatewayRoutePathMatchConfig {
+  /**
+   * Gateway route configuration for matching on the complete URL path of the request.
+   *
+   * @default - no matching will be performed on the complete URL path
+   */
+  readonly wholePathMatch?: CfnGatewayRoute.HttpPathMatchProperty;
+
+  /**
+   * Gateway route configuration for matching on the prefix of the URL path of the request.
+   *
+   * @default - no matching will be performed on the prefix of the URL path
+   */
+  readonly prefixPathMatch?: string;
+
+  /**
+   * Gateway route configuration for rewriting the complete URL path of the request..
+   *
+   * @default - no rewrite will be performed on the request's complete URL path
+   */
+  readonly wholePathRewrite?: CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty;
+
+  /**
+   * Gateway route configuration for rewriting the prefix of the URL path of the request.
+   *
+   * @default - rewrites the request's URL path to '/'
+   */
+  readonly prefixPathRewrite?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Defines HTTP gateway route matching based on the URL path of the request.
+ */
+export abstract class HttpGatewayRoutePathMatch {
+  /**
+   * The value of the path must match the specified value exactly.
+   * The provided `path` must start with the '/' character.
+   *
+   * @param path the exact path to match on
+   * @param exactPathRewrite the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static exactly(path: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ exact: path }, { exact: exactPathRewrite });
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   *
+   * @param regex the regex used to match the path
+   * @param exactPathRewrite the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static regex(regex: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ regex }, { exact: exactPathRewrite });
+  }
+
+  /**
+   * The value of the path must match the specified prefix.
+   *
+   * @param prefixPathMatch the value to use to match the beginning of the path part of the URL of the request.
+   *   When `rewriteTo` is provided, `prefixPathMatch` must start and end with the '/' character.
```

Comment:
Can we please validate these conditions? (Plus unit tests checking those validations)

Reply:
We will be reverting back all validations.